lib/phoenix/controller/pipeline.ex

```diff
@@ -125,11 +126,40 @@ defmodule Phoenix.Controller.Pipeline do
         _ = var!(controller)
         _ = var!(action)
 
-        unquote(body)
+        try do
+          unquote(body)
+        catch
+          kind, reason ->
+            Phoenix.Controller.Pipeline.__catch__(
```

Comment:
Because this is wrapping the whole pipeline. The chance of false positives can be big. Maybe we should override `action/2` only?